test/common/unit/net/test_bolt1.py

```diff
@@ -0,0 +1,385 @@
+    mtype = b"\x00\x10"
+    global_features = b"\x2a\xaa\xaa"  # All odd
+    gflen = b"\x00\x03"
+    local_features = b"\x01"  # Feature 1 even
+    flen = b"\x00\x01"
+    im2 = InitMessage.from_bytes(mtype + gflen + global_features + flen + local_features)
+    assert (
+        isinstance(im2, InitMessage)
+        and im2.type == mtype
+        and im2.global_features.serialize() == global_features
+        and im2.local_features.serialize() == local_features
+    )
+
+    # With some networks
+    networks = NetworksTLV([get_random_value_hex(32) for _ in range(5)])
+    im3 = InitMessage.from_bytes(mtype + gflen + global_features + flen + local_features + networks.serialize())
+    assert (
+        isinstance(im3, InitMessage)
+        and im3.type == mtype
+        and im3.global_features.serialize() == global_features
+        and im3.local_features.serialize() == local_features
+        and im3.networks == networks
+    )
+
+
+def test_init_message_from_bytes_wrong():
+    # Message is not bytes
+    with pytest.raises(TypeError, match="message be must a bytearray"):
+        InitMessage.from_bytes("message")
+
+    # Message is not long enough < 6
+    with pytest.raises(ValueError, match="message be must at least 6-byte long"):
+        InitMessage.from_bytes(b"\x00\x10\x00")
+
+    # Type is not init
+    with pytest.raises(ValueError, match="Wrong message format. types do not match"):
+        InitMessage.from_bytes(b"\x00\x00\x00\x00\x00\x00")
+
+    # Encoded lengths are wrong causing an unexpected EOF
+    with pytest.raises(ValueError, match="Wrong message format. Unexpected EOF"):
+        InitMessage.from_bytes(b"\x00\x10\x00\x01\x00\x00")
+
+
+def test_error_message():
+    # Error message expects a channel_id (32-hex encoded str) and an optional data field
+    cid = get_random_value_hex(32)
+    em = ErrorMessage(cid)
+    assert isinstance(em, ErrorMessage)
+    assert em.channel_id == cid
+    assert em.data is None
+
+    # Same with associated data
+    data = "error message data"
+    em2 = ErrorMessage(cid, data)
+    assert isinstance(em, ErrorMessage)
+    assert em2.channel_id == cid
+    assert em2.data == data
+
+
+def test_error_message_wrong():
+    # Channel id must be a 32-byte hex str
+    # Data must be string if set and no longer than the message cap size when encoded pow(2, 16)
+
+    # Wrong channel id
+    with pytest.raises(ValueError, match="channel_id must be a 256-bit hex string"):
+        ErrorMessage(get_random_value_hex(31))
+
+    with pytest.raises(ValueError, match="channel_id must be a 256-bit hex string"):
+        ErrorMessage(dict())
+
+    # Wrong data type
+    with pytest.raises(ValueError, match="data must be string if set"):
+        ErrorMessage(get_random_value_hex(32), b"message")
+
+    # Data too long
+    with pytest.raises(ValueError, match=f"Encoded data length cannot be bigger than {pow(2, 16)}"):
+        ErrorMessage(get_random_value_hex(32), "A" * (pow(2, 16) + 1))
```

Comment:
Since data length is encoded in a `u16`, the maximum length should be `pow(2, 16) - 1`, meaning that `pow(2, 16)` is already too long.